### ISSUE TITLE
Package commands fail with error "Flag expects a value"

### ISSUE THREAD

**Issue by chris-duncan-adaptiva** [forcedotcom/cli#2302]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Summary
Executing package commands fails with `Error (1): Flag --<foo> expects a value` (<foo> is a placeholder for the actual flag).

For example, `sf package create -n MyPackage -t Unlocked -r force-app` fails with `Error (1): Flag --package-type expects a value`.

For another example,  `sf package version create -p MyPackage` for a package aliased as "MyPackage" fails with `Error (1): Flag --package expects a value`. However, for some reason, executing the same command with the package ID (e.g. `sf package version create -p 0Hoxxxxxxxxxxxxxxx`) works OK.

### Steps To Reproduce

> **IMPORTANT**
> Provide a repository that's configured to reproduce the issue. If you are unable to provide a repo, please explain why not. The more info we have from the start, the faster we can resolve your issue.
> We may close your issue if you don't include proper instructions.
>
> - Generate a project with `sf project generate` or fork [dreamhouse-lwc](https://github.com/trailheadapps/dreamhouse-lwc).
> - Provide detailed step-by-step instructions on how to reproduce the issue.

Fork: https://github.com/chris-duncan-adaptiva/dreamhouse-lwc-2302
1. Navigate to dreamhouse root /bin and execute install script.
2. Or, for a different project/repo, manually create a package by executing `sf package create -n MyPackage -t Unlocked -r force-app`.

### Expected result
Command begins creating a new package using the specified package details.

### Actual result
Command fails with `Error (1): Flag --package-type expects a value` even though a value for the flag is provided.

### System Information

bash, pwsh7

```json
{
  "cliVersion": "@salesforce/cli/2.0.2",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.16.1",
  "osVersion": "Windows_NT 10.0.22621",
  "shell": "cmd.exe",
  "rootPath": "C:\\Users\\ChrisDuncan\\AppData\\Roaming\\nvm\\v18.16.1\\node_modules\\@salesforce\\cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.1 (core)",
    "@oclif/plugin-commands 2.2.17 (core)",
    "@oclif/plugin-help 5.2.11 (core)",
    "@oclif/plugin-not-found 2.3.26 (core)",
    "@oclif/plugin-plugins 3.1.6 (core)",
    "@oclif/plugin-search 0.0.18 (core)",
    "@oclif/plugin-update 3.1.22 (core)",
    "@oclif/plugin-version 1.3.6 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.41 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/cli 2.0.2 (core)",
    "apex 2.3.4 (core)",
    "auth 2.8.4 (core)",
    "custom-metadata 2.1.23 (user)",
    "data 2.4.5 (core)",
    "deploy-retrieve 1.13.5 (core)",
    "env 2.1.14 (user)",
    "functions 1.21.10 (user)",
    "info 2.6.25 (core)",
    "limits 2.3.23 (core)",
    "login 1.2.15 (core)",
    "org 2.9.19 (core)",
    "packaging 1.20.1 (user)",
    "schema 2.3.17 (core)",
    "settings 1.4.17 (core)",
    "signups 1.4.22 (user)",
    "sobject 0.1.32 (core)",
    "source 2.10.22 (core)",
    "telemetry 2.2.1 (core)",
    "templates 55.5.2 (core)",
    "trust 2.4.27 (core)",
    "user 2.3.20 (core)"
  ]
}
```

### Additional information
sf doctor
```json
{
  "versionDetail": {
    "cliVersion": "@salesforce/cli/2.0.2",
    "architecture": "win32-x64",
    "nodeVersion": "node-v18.16.1",
    "osVersion": "Windows_NT 10.0.22621",
    "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
    "rootPath": "C:\\Users\\ChrisDuncan\\AppData\\Roaming\\nvm\\v18.16.1\\node_modules\\@salesforce\\cli",
    "pluginVersions": [
      "@oclif/plugin-autocomplete 2.3.1 (core)",
      "@oclif/plugin-commands 2.2.17 (core)",
      "@oclif/plugin-help 5.2.11 (core)",
      "@oclif/plugin-not-found 2.3.26 (core)",
      "@oclif/plugin-plugins 3.1.6 (core)",
      "@oclif/plugin-search 0.0.18 (core)",
      "@oclif/plugin-update 3.1.22 (core)",
      "@oclif/plugin-version 1.3.6 (core)",
      "@oclif/plugin-warn-if-update-available 2.0.41 (core)",
      "@oclif/plugin-which 2.2.22 (core)",
      "@salesforce/cli 2.0.2 (core)",
      "apex 2.3.4 (core)",
      "auth 2.8.4 (core)",
      "custom-metadata 2.1.23 (user)",
      "data 2.4.5 (core)",
      "deploy-retrieve 1.13.5 (core)",
      "env 2.1.14 (user)",
      "functions 1.21.10 (user)",
      "info 2.6.25 (core)",
      "limits 2.3.23 (core)",
      "login 1.2.15 (core)",
      "org 2.9.19 (core)",
      "packaging 1.20.1 (user)",
      "schema 2.3.17 (core)",
      "settings 1.4.17 (core)",
      "signups 1.4.22 (user)",
      "sobject 0.1.32 (core)",
      "source 2.10.22 (core)",
      "telemetry 2.2.1 (core)",
      "templates 55.5.2 (core)",
      "trust 2.4.27 (core)",
      "user 2.3.20 (core)"
    ]
  },
  "sfdxEnvVars": [],
  "sfEnvVars": [
    [
      "SF_AUTOUPDATE_DISABLE",
      "true"
    ],
    [
      "SF_DISABLE_AUTOUPDATE",
      "true"
    ],
    [
      "SF_UPDATE_INSTRUCTIONS",
      "Use \"npm update --global @salesforce/cli\" to update npm-based installations."
    ]
  ],
  "cliConfig": {
    "_base": "@oclif/core@2.8.11",
    "debug": 0,
    "topicSeparator": " ",
    "warned": false,
    "commandPermutations": {
      "validPermutations": {}
    },
    "topicPermutations": {
      "validPermutations": {}
    },
    "_commands": {},
    "_topics": {},
    "root": "C:\\Users\\ChrisDuncan\\AppData\\Roaming\\nvm\\v18.16.1\\node_modules\\@salesforce\\cli",
    "name": "@salesforce/cli",
    "version": "2.0.2",
    "channel": "stable",
    "valid": true,
    "arch": "x64",
    "platform": "win32",
    "windows": true,
    "bin": "sf",
    "binAliases": [
      "sfdx"
    ],
    "nsisCustomization": "scripts/nsis.nsi",
    "dirname": "sf",
    "flexibleTaxonomy": true,
    "userAgent": "@salesforce/cli/2.0.2 win32-x64 node-v18.16.1",
    "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
    "home": "C:\\Users\\ChrisDuncan",
    "cacheDir": "C:\\Users\\ChrisDuncan\\AppData\\Local\\sf",
    "configDir": "C:\\Users\\ChrisDuncan\\AppData\\Local\\sf",
    "dataDir": "C:\\Users\\ChrisDuncan\\AppData\\Local\\sf",
    "errlog": "C:\\Users\\ChrisDuncan\\AppData\\Local\\sf\\error.log",
    "nodeEngine": ">=14.0.0"
  },
  "pluginSpecificData": {
    "@salesforce/plugin-source": [
      {
        "sourceApiVersion": "58.0",
        "targetDevHubApiVersion": "58.0"
      }
    ]
  },
  "diagnosticResults": [
    {
      "testName": "salesforcedx plugin not installed",
      "status": "pass"
    },
    {
      "testName": "no linked plugins",
      "status": "pass"
    },
    {
      "testName": "[@salesforce/plugin-source] sourceApiVersion matches apiVersion",
      "status": "pass"
    },
    {
      "testName": "using latest or latest-rc CLI version",
      "status": "pass"
    }
  ],
  "suggestions": [
    "Check https://github.com/forcedotcom/cli/issues for CLI issues posted by the community.",
    "Check http://status.salesforce.com for general Salesforce availability and performance."
  ],
  "logFilePaths": [],
  "commandExitCode": 0
}
```

**Comment by mshanemc** [comment]
that looks like a bug that happened in oclif.  We've [got it fixed there](https://github.com/oclif/core/pull/734), and will need to pull in those changes.

As a workaround, can you try `sf plugins uninstall @salesforce/plugin-packaging` and then re-install it (or just re-run some command)?  I think part of the problem is that just-in-time plugins pull in the very latest of all the dependencies, rather than the ones the CLI shipped with.

**Comment by chris-duncan-adaptiva** [comment]
Your workaround worked for me. Thank you!

**Comment by chris-duncan-adaptiva** [comment]
Appears to be fixed in 2.1.7, thanks!
